src/policy/space.rs

```diff
@@ -524,7 +522,32 @@ impl<VM: VMBinding> CommonSpace<VM> {
         // FIXME
         rtn.descriptor = SpaceDescriptor::create_descriptor_from_heap_range(start, start + extent);
         // VM.memory.setHeapRange(index, start, start.plus(extent));
-        args.plan_args.vm_map.insert(start, extent, rtn.descriptor);
+
+        // We only initialize our vm map if the range of the space is in our available heap range. For normally spaces,
+        // they are definitely in our heap range. But for VM space, a runtime could give us an arbitrary range. We only
+        // insert into our vm map if the range overlaps with our heap.
+        {
+            use crate::util::heap::layout;
+            if layout::range_overlaps_available_range(start, extent) {
+                // Crop the range to our available heap range. We only need to set VM map for the range that is in our heap
+                let start_in_range = if start > layout::vm_layout_constants::AVAILABLE_START {
+                    start
+                } else {
+                    layout::vm_layout_constants::AVAILABLE_START
+                };
+                let end_in_range = if start + extent < layout::vm_layout_constants::AVAILABLE_END {
+                    start + extent
+                } else {
+                    layout::vm_layout_constants::AVAILABLE_END
+                };
```

Comment:
This piece of code is somewhat hard to read because they span a few lines and involves multiple variables, although the logic is simple (using the max of two lower bounds as the lower bound, and using the min of two upper bounds as the upper bound). We could use `max` and `min` methods instead of the `if` statements.  But I suggest using `Range` (i.e. the `a..b` expression) for this purpose.  See my comments on the `range_overlaps_available_range` method.